Objects/typeobject.c

```diff
@@ -2320,8 +2331,39 @@ _PyType_CalculateMetaclass(PyTypeObject *metatype, PyObject *bases)
     return winner;
 }
 
+
+static int
+type_set_default_module(PyObject *dict, int depth)
+{
+    if (_PyDict_GetItemIdWithError(dict, &PyId___module__) != NULL) {
+        return 0;
+    }
+    if (PyErr_Occurred()) {
+        return -1;
+    }
+
+    PyFrameObject *f = PyEval_GetFrame();
+    if (f == NULL) {
+        return 0;
+    }
+    while (depth-- > 0 && f->f_back) {
+        f = f->f_back;
+    }
```

Comment:
Unless I am missing something, `depth` can be only either 0 or 1, maybe it is better to use a name like `skip` or `nested` and write this as:
```c
if (skip_frame && f->f_back) {
    f = f->f_back;
}
```